db/seeds.rb

```diff
@@ -299,16 +300,23 @@ def generate_reservation
                              checkout_length.days).to_date
     res.notes = FFaker::HipsterIpsum.paragraph(8)
     res.notes_unsent = [true, false].sample
-    if rand < PAST_CHANCE
-      throw_into_past res
-      return
+
+    throw_into_past res if rand < PAST_CHANCE
+
+    # TODO: remove
+    if res.overdue && res.start_date > Time.zone.today
+      puts "GENERATED A FUTURE OVERDUE RES"
     end
     try_again = false
     begin
       res.save!
     rescue ActiveRecord::RecordInvalid
       try_again = true
     end
+    # TODO: remove
+    if res.overdue && res.start_date > Time.zone.today && !try_again
+      puts "SAVED A FUTURE OVERDUE RES"
+    end
```

Comment:
In 100 runs, generating 100 reservations per run, 50 future overdue reservations were created, but none were saved. These test logging statements will be removed in the final squash, but I've left them for local testing.